metadata/src/main/java/org/apache/kafka/metadata/migration/KRaftMigrationDriver.java

```diff
@@ -174,12 +173,12 @@ private boolean areZkBrokersReadyForMigration() {
     /**
      * Apply a function which transforms our internal migration state.
      *
-     * @param name  A descriptive name of the function that is being applied
-     * @param stateMutator  A function which performs some migration operations and possibly transforms our internal state
+     * @param name         A descriptive name of the function that is being applied
+     * @param migrationOp  A function which performs some migration operations and possibly transforms our internal state
      */
-    private void apply(String name, Function<ZkMigrationLeadershipState, ZkMigrationLeadershipState> stateMutator) {
+    private void applyMigrationOperation(String name, KRaftMigrationOperation migrationOp) {
         ZkMigrationLeadershipState beforeState = this.migrationLeadershipState;
-        ZkMigrationLeadershipState afterState = stateMutator.apply(beforeState);
+        ZkMigrationLeadershipState afterState = migrationOp.apply(beforeState);
         log.trace("{} transitioned from {} to {}", name, beforeState, afterState);
```

Comment:
I feel like if the state is meaningfully different, we'd want this to be higher than TRACE.

Like the metadata offset change is fine to log at TRACE, but if the controller epoch changes or controller ID changes, wouldn't we want this to be INFO?

I guess we could do this in a follow-on PR as well

Reply:
I'll add this to the current PR. Shouldn't be too hard